File: spoj/basics/java/CPTTRN1-Character_Patterns_Act_3.java
import java.util.*;
import java.lang.*;

class Main {
  public static boolean oddOrNot(int num) { return (num & 1) != 0; }

  public static void main (String[] args) throws java.lang.Exception {
    Scanner scan = new Scanner(System.in);
    String separator, str_line, str_colun = "..*";

    int t = Integer.parseInt(scan.nextLine()), l, c;
    for (int i = 0; i < t; ++i) {
      l = scan.nextInt();
      c = scan.nextInt();

      str_line  = "*";
      separator = "*";
      for (int j = 0; j < c; ++j) { separator += "***"; str_line += str_colun; }

      str_line += "\n" + str_line;
      for (int j = 0; j < l; ++j) { System.out.println(separator); System.out.println(str_line); }

      System.out.println(separator);
    }
  }
}

File: spoj/basics/c++/CPTTRN1-Character_Patterns_Act_3.cpp
#include <iostream>

using namespace std;

int main() {
	int t, l, c;
	string separator, str_colun, str_line;

	cin >> t;
	str_colun = "..*";
	for (int i = 0; i < t; ++i) {
		cin >> l >> c;

		str_line  = "*";
		separator = "*";
		for (int i = 0; i < c; ++i) { separator += "***"; str_line += str_colun; }

		str_line+= "\n" + str_line;
		for (int i = 0; i < l; ++i) { cout << separator << endl << str_line << endl; }

		cout << separator << endl;
	}

	return 0;
}
